### PR TITLE
Allow as json serialization

### PR DESCRIPTION
**Pull request by erose**
After this PR, `response.to_json` should yield a good representation of the parsed data as JSON.

Not a priority to review (not blocking anything).

### CODE REVIEW

**Comment by marisawallace** [comment]
@erose FYI, I made a small logic change in the PR, added a lot of docs.

File: lib/typeform_data/value_class.rb [hunk 1]
```diff
@@ -23,6 +23,49 @@ def initialize(config, attrs)
...
+    #
+    # One last thing: if you pass options #to_json or #as_json, Rails will use the same options for
+    # all calls in the object graph, so the following behavior is expected (if counterintuitive):
+    #
+    #    #   response.as_json(only: 'answers')
+    #   => {"answers"=>[{}, {}, {}, {}]}
+    #
+    # @param [Hash]
+    def as_json(options = {})
+      # Note: :config doesn't work here-- using the string form is mandatory.
```
Comment (erose) [inline lib/typeform_data/value_class.rb:63]: Thanks, this was a big gotcha.

File: lib/typeform_data/value_class.rb [hunk 2]
```diff
@@ -23,6 +23,49 @@ def initialize(config, attrs)
...
+      return super({ except: 'config' }) if options.empty?
+
+      super(options.merge(except: ['config'] + Array(options[:except])))
```
Comment (erose) [inline lib/typeform_data/value_class.rb:66]: Thanks for the correctness change. This is and the above line amount to a a https://apidock.com/rails/Hash/deep_merge just for the `except` key, right? I'd wonder if there's a way to express it in those terms -- but probably not worth it. No action required.

Comment (marisawallace) [reply lib/typeform_data/value_class.rb:66]: @erose correct. And I agree!

**Pull request merged**